### PR TITLE
Patch 2

### PR DESCRIPTION
**Pull request by lilulilulilu**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by lilulilulilu** [comment]
fix the bug: cannot set custom similarity function while using custom embedding model. so that I can pass a similarity_fn to VectorIndexRetriever as below:
    # configure retriever
    retriever = VectorIndexRetriever(
        index=index, 
        similarity_top_k=2,
        similarity_fn = embed_model.similarity
    )

File: llama_index/indices/vector_store/retrievers/retriever.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def __init__(
         self._doc_ids = doc_ids
         self._filters = filters
 
-        self._kwargs: Dict[str, Any] = kwargs.get("vector_store_kwargs", {})
+        self._kwargs: Dict[str, Any] = kwargs
```
Comment (logan-markewich) [inline llama_index/indices/vector_store/retrievers/retriever.py:61]: This seems like a breaking change. `vector_store_kwargs` was how you were able to pass filters and other parameters

**Comment by lilulilulilu** [comment]
then How to pass custom similarty function?



---Original---
From: ***@***.***&gt;
Date: Sat, Jul 8, 2023 00:21 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [jerryjliu/llama_index] Patch 2 (PR #6769)




 
@logan-markewich commented on this pull request.
 
 
In llama_index/indices/vector_store/retrievers/retriever.py:
 &gt; @@ -58,7 +58,7 @@ def __init__(          self._doc_ids = doc_ids          self._filters = filters   -        self._kwargs: Dict[str, Any] = kwargs.get("vector_store_kwargs", {}) +        self._kwargs: Dict[str, Any] = kwargs  
This seems like a breaking change. vector_store_kwargs was how you were able to pass filters and other parameters

**Comment by logan-markewich** [comment]
@lilulilulilu maybe the similarity function should be passed in using the current `vector_store_kwargs` then?

**Comment by Disiok** [comment]
Closing since inactive

**Pull request closed without merging**